app/routes.py

```diff
@@ -1,2 +1,51 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request
 
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+@planets_bp.route("", methods=["GET", "POST"])
+def handle_planets():
+    if request.method == "GET":
+        name_query = request.args.get("name")
+        if name_query:
+            planets = Planet.query.filter_by(name=name_query)
+        else:
+            planets = Planet.query.all()
+        planets_response = [planet.to_json() for planet in planets]
+
+        return jsonify(planets_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        new_planet = Planet(
+            name=request_body["name"], description=request_body["description"]
+        )
+
+        db.session.add(new_planet)
+        db.session.commit()
+
+        return make_response(f"Planet {new_planet.name} successfully created", 201)
+
+
+@planets_bp.route("/<planet_id>", methods=["GET", "PUT", "DELETE"])
+def handle_planet(planet_id):
+    planet = Planet.query.get_or_404(planet_id)
```

Comment:
Good use of `get_or_404`. If you were to separate out `handle_planet` into different functions for each verb, consider how you could move this functionality into a helper function.

Reply:
Consider also how this method would behave if `planet_id` was not an integer. How can we build in error checking for the parameters in the route.